### ISSUE TITLE
CM12.1 Preparations

### ISSUE THREAD

**Issue by JustArchi** [ArchiDroid/android_device_samsung_i9300#73]
@ArchiDroid/developers 

If possible, we should commit everything we have in non-device branches (android_*) to CM's gerrit, if it's not already there.

If it's not possible, we should start fresh from cm-12.1 branch (when it's available), and then re-commit everything we need on top of that. Don't start until actually there will be cm-12.1 branch, as CM is planning a cleanup, and we'd need to do that cleanup as well.

```
git fetch https://github.com/CyanogenMod/android_build cm-12.1
git checkout FETCH_HEAD
git checkout -b cm-12.1
(cherry-picks here if needed)
git push https://github.com/ArchiDroid/android_build cm-12.1
```

Device branches will be just a new branch with old content.

```
git checkout -b cm-12.1
git push https://github.com/ArchiDroid/android_device_samsung_i9300 cm-12.1
```

This way we won't lose anything, and it'll be nice and tidy again.

**Comment by arter97** [comment]
Will push a cm-12.1 branch soon with kanged patch-sets applied.
But just like how staging/cm-12.1 on github.com/CyanogenMod is, repos may be force-updated.

**Comment by JustArchi** [comment]
It's pointless to push staging/cm-12.1, but if you want...

**Comment by arter97** [comment]
I'm thinking of making a one giant kanged commit instead of dozens of patch as we'll have to fix conflicts more.
This will make future updates more easy even when CM force-updates their Git.

git checkout -b cm-12.1
git reset --hard upstream/cm-12.1
git diff upstream/cm-12.0..origin/cm-12.0 | patch -p1

What do you think?

**Comment by arter97** [comment]
(And btw, I've already applied those git patches on staging/cm-12.1 with the experimental build I've done earlier)

**Comment by JustArchi** [comment]
As I pointed above, ideal situation is that we don't need to apply ANY patches to non-device trees at all. They should be on CM's gerrit, and submitted to upstream.

**Comment by JustArchi** [comment]
But sure, it's staging branch anyway, so you can squash everything and make it compile, just don't forward it to cm-12.1 when it's alive :+1:

**Comment by arter97** [comment]
I've seen some updates to 4x12 on CyanogenMod/cm-12.0, but how's it really going?
And what do you mean by "don't forward it to cm-12.1"? :S

**Comment by JustArchi** [comment]
I mean that non-device trees (such as android_build), if they include any i9300 patches from us, should be sent to CM's gerrit, instead of being cherry-picked by us.

We should get rid of everything here -> https://github.com/ArchiDroid/manifest/blob/cm-12.0/roomservice.xml#L3

Device trees are another story, as we don't need to touch them at all.

**Comment by JustArchi** [comment]
And I meant that once CM make cm-12.1 branch, we shouldn't squash anything and instead cherry-pick what we need, because we must keep track of every fix.

**Comment by arter97** [comment]
bionic, av and native seems dangerous.
Is CM interested in merging our changes?

**Comment by JustArchi** [comment]
We'll never get any official build if they're not interested in frameworks.

**Comment by arter97** [comment]
Yup, I'm aware of that of course :P 
What I'd like to know is are they interested in merging Exynos changes on official Git repos...

**Comment by JustArchi** [comment]
They should merge everything, as long as we're not destroying other devices.

**Comment by arter97** [comment]
Sounds promising, but then how come there isn't any official i9300 build?
Haven't we sent every patches yet?

**Comment by JustArchi** [comment]
Because there is no maintainer...

**Comment by arter97** [comment]
They can kidnap Moster2 for the job, can't they?
I've seen many of my fellow unofficial CM porters becoming official maintainers.

**Comment by JustArchi** [comment]
Yeah, but not sure if Moster want to be kidnapped :dancer:

**Comment by arter97** [comment]
XD

**Comment by MarcKe** [comment]
Hehe, very funny :dancers:

**Comment by arter97** [comment]
You deserves to get kidnapped (lol)

**Comment by JustArchi** [comment]
https://github.com/CyanogenMod/android/tree/cm-12.1

Ready, set, go.

**Comment by JustArchi** [comment]
I set up everything on new cm-12.1 branches. @MarcKe do you want to cherry-pick missing stuff in base repos? I know that CM12 is your toy and you probably want to have control over what is missing and should be uploaded to CM's gerrit :+1:.

If you don't have time/willings, I'll do the job.

**Comment by MarcKe** [comment]
No time, sry

**Comment by JustArchi** [comment]
Alright :+1:

**Comment by JustArchi** [comment]
Preparations done, tree is in buildable state. Done by me :+1:
